persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/JdbcBasePersistenceImpl.java

```diff
@@ -269,8 +291,7 @@ public PolarisBaseEntity lookupEntity(
       @Nonnull PolarisCallContext callCtx, long catalogId, long entityId, int typeCode) {
     Map<String, Object> params =
         Map.of("catalog_id", catalogId, "id", entityId, "type_code", typeCode, "realm_id", realmId);
-    String query = generateSelectQuery(new ModelEntity(), params);
-    return getPolarisBaseEntity(query);
+    return getPolarisBaseEntity(queryGenerator.generateSelectQuery(new ModelEntity(), params));
```

Comment:
Not related to this change, but why do we need so many WHERE conditions? The PK is `(realmId, id)`, so that should be sufficient.